src/TaskQueue.js

```diff
@@ -0,0 +1,77 @@
+const iterate = require('taskcluster-lib-iterate');
+const assert = require('assert');
+const _ = require('lodash');
+class TaskQueue {
+  constructor(cfg, queue) {
+    assert(cfg.worker.workerId, 'Worker ID is required');
+    assert(cfg.worker.workerType, 'Worker type is required');
+    assert(cfg.worker.workerGroup, 'Worker group is required');
+    assert(cfg.worker.provisionerId, 'Provisioner ID is required');
+    assert(queue, 'Instance of taskcluster queue is required');
+    this.queue = queue;
+    this.workerType = cfg.worker.workerType;
+    this.provisionerId = cfg.worker.provisionerId;
+    this.workerGroup = cfg.worker.workerGroup;
+    this.workerId = cfg.worker.workerId;
+  }
+  async startWorker() {
+
+    while (true) {
+      var res = await this.claimTasks();
+    }
+  }
+
+  async claimTasks() {
+    var capacity = 1;
+    let result = await this.queue.claimWork(this.provisionerId, this.workerType, {
+      tasks: capacity,
+      workerGroup: this.workerGroup,
+      workerId: this.workerId,
+    });
+    let stat = '';
+    if (_.isEmpty(result.tasks.task.payload)) {
```

Comment:
Probably no need for `lodash` here. If `result.tasks.task.payload` is an array, you can check its length, if it's an object, you can test `Object.keys(result.tasks.task.payload).length` (in fact, this test will work for arrays as well)